python/tests/test_mesh_metric.py

```diff
@@ -0,0 +1,28 @@
+import sys
+import os
+sys.path.append(os.path.abspath('..'))
+from dolfin import *
+from adaptivity import *
+
+check = []
+
+for n in range(2, 20):
+    mesh = UnitSquareMesh(mpi_comm_world(), n,n)
+
+    M = mesh_metric(mesh)
+
+    A = M.vector().array().reshape((mesh.num_cells(),2,2))
+
+    # the metric should be constant on each element
+    if abs(A - A[0,:,:]).max()/A.max() > 1.0e-14:
+        raise RuntimeError('Test failed: the metric is not constant on each element')
+
```

Comment:
Here, can you please add a check : the metric should actually be equal to  1/h^2 * [[1,0.5],[0.5],1]] ? Thanks!

Reply:
I can, but the mesh_metric code outputs -0.5 in the off-diagonals. Is this correct?

If not we should replace the mesh_metric code with the exact formula you highlighted above.

Reply:
I can confirm that -0.5 on the off-diagonals is wrong, but I am certain that mesh_metric used to produce the correct output. Either the code itself or FEniCS has changed.

Reply:
the -0.5 is correct, I double checked it with the exact formula. That part of the mesh_metric code did not change and not even FEniCS (but the code was correct anyways :) )